### PR TITLE
made pandora validate credentials in configuration

### LINKED ISSUE #720: Validate account credentials during stream instantiation
Test stream creds during instantiation, probably in the constructor in python.

### PR DESCRIPTION
**Pull request by klay2000**
### Checklist

* [x] Have you tested your changes and ensured they work?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] If applicable, have you updated the CHANGELOG?
* [x] Does your submission pass linting & tests? You can test on localhost using `./scripts/test`

closes #720

### CODE REVIEW

File: v1
Comment (linknum23) [inline v1:1]: oops

Comment (klay2000) [reply v1:1]: oop

**Review by rtertiaer — APPROVED** [review]
seems legit, except for the hour+ github action. if you can replicate a check that doesn't last hours when github isn't seemingly having a bad hairday, i'd ship it

File: amplipi/streams.py [hunk 1]
```diff
@@ -715,6 +715,9 @@ def validate_stream(self, **kwargs):
     if 'name' in kwargs and not re.fullmatch(HOSTNAME_LIKE, kwargs['name']):
       raise InvalidStreamField("name", "invalid device name")
 
+# TODO: A significant amount of complexity could be removed if we switched some features here to using pydora instead of
+# interfacing with pianobar's TUI
```
Comment (rtertiaer) [inline amplipi/streams.py:719]: :100: :100: :100: :100: :100: :100:

**Review by rtertiaer — APPROVED** [review]

**Pull request merged**